### PR TITLE
Issue/1517

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1517 

## How Has This Been Tested?
I have tested this with 4 types of donation forms:
1. Donation Form with Guest Checkout enabled and no registration and login.
2. Donation Form with Guest Checkout enabled and registration enabled.
3. Donation Form with Guest Checkout disabled and registration enabled.
4. Donation Form with Guest Checkout enabled and registration as well as login enabled.

@DevinWalker @ravinderk According to the current scenario, if we disable guest checkout and enable registration then the registration fields become `required`. But, in this PR, the same scenario is not added as i think that adding required field to checkbox `Create an account` is not good. Let me know your thoughts.

## Screenshots (jpeg or gifs if applicable):
image

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/emails/class-give-email-tags.php [hunk 1]
```diff
@@ -1223,3 +1230,78 @@ function __give_20_bc_str_type_email_tag_param( $tag_args ) {
...
+ *
+ * @param array $tag_args Array of arguments for email tags.
+ *
+ * @since 2.0
+ *
+ * @return array
+ */
+function give_email_tag_reset_password_link( $tag_args ) {
+
+	$reset_password_url = give_get_reset_password_url(
```
Comment (ravinderk) [inline includes/emails/class-give-email-tags.php]: @mehul0810 Add support for donor id here. check how I am using switch case in email tag render functions.

**Comment by ravinderk** [comment]
@mehul0810 
1. Please write an automatic update to insert `reset_password_link ` in `Donor Register` email message if it has value non-empty.

If `Give` already setup on a site then there are chances that `Donor Register` email message set to previous default email message which does not has `reset_password_link `, so donor will not get this link which creates lots of support request.

Let me know if you have any question regarding that.

File: includes/admin/upgrades/upgrade-functions.php [hunk 2]
```diff
@@ -1043,6 +1044,23 @@ function give_v20_upgrades_email_setting() {
...
+ */
+function give_v20_upgrade_donor_register_email_message() {
+	$all_setting = give_get_settings();
+
+	// Bailout on fresh install.
+	if ( empty( $all_setting ) ) {
+		return;
+	}
+
+	$donor_register_email = new Give_Donor_Register_Email();
```
Comment (ravinderk) [inline includes/admin/upgrades/upgrade-functions.php]: @mehul0810 Use `get_instance` function here

Comment (mehul0810) [reply includes/admin/upgrades/upgrade-functions.php]: @ravinderk I've updated the code. Please review,.

File: includes/forms/template.php [hunk 3]
```diff
@@ -1139,50 +1207,40 @@ function give_get_register_fields( $form_id ) {
...
+			<div id="give-create-account-wrap-<?php echo $form_id; ?>" class="form-row form-row-first form-row-responsive">
+				<label for="give-create-account-<?php echo $form_id; ?>">
+					<?php
+					// Add attributes to checkbox, if Guest Checkout is disabled.
+					$is_guest_checkout = give_get_meta( $form_id,'_give_logged_in_only', true );
+					if( ! give_is_setting_enabled( $is_guest_checkout ) ) {
+						echo Give()->tooltips->render(
+							array(
+								'tag_content' => '<input type="checkbox" name="give_create_account" id="give-create-account-" class="give-input" checked disabled /><input type="hidden" name="give_create_account" value="on" />',
+								'label' => __( 'Registration is compulsory for this donation form as Guest Checkout is disabled.', 'give' ),
```
Comment (ravinderk) [inline includes/forms/template.php]: @DevinWalker Do you want to change this message?

Comment (DevinWalker) [reply includes/forms/template.php]: Yes. It should be `Registration is required to donate.`

**Pull request merged**